lib/apiservers/engine/backends/backends.go

```diff
@@ -90,13 +93,34 @@ func Init(portLayerAddr, product string, config *config.VirtualContainerHostConf
 		return err
 	}
 
+	var wg sync.WaitGroup
+
+	initNum := 2
+	wg.Add(initNum)
+	results := make(chan error, initNum)
+
 	log.Info("Refreshing image cache")
 	go func() {
+		defer wg.Done()
 		if err := cache.ImageCache().Update(portLayerClient); err != nil {
-			log.Warn("Failed to refresh image cache: %s", err)
+			err = errors.Errorf("Failed to refresh image cache: %s", err)
+			log.Warn(err)
+			results <- err
```

Comment:
@hmahmood so you think it should still be a warning, instead of error?
But in theory, the error only happens while portlayer has problem, e.g. query from vsphere. At this moment, I think it might be better to stop the initialization.